src/Repository/ReaderRepository.php

```diff
@@ -2,26 +2,87 @@
 
 namespace BookReviews\Repository;
 
+use \PDO;
+
 /**
  * Class ReaderRepository
  * @package BookReview\Repository
  */
-class ReaderRepository extends DatabaseConnection
+class ReaderRepository implements Repository
 {
     /**
-     * @param string $username
-     * @param string $password
+     * @var PDO
      */
-    public function insertIntoDatabase($username, $password)
+    private $conn;
+
+    public function __construct()
+    {
+        $database = new DatabaseConnectionBuilder();
+        $this->conn = $database->connectToDatabase();
+    }
+
+    public function getId()
+    {
+        // TODO: Implement getId() method.
+    }
+
+    public function add()
+    {
+        // TODO: Implement add() method.
+    }
+
+    public function update()
+    {
+        // TODO: Implement update() method.
+    }
+
+    public function delete()
     {
-         die();
+        // TODO: Implement delete() method.
     }
     /**
+     * The function returns  true if the result of the query returns an array
+     * and the password matches the one in the array.
      * @param string $username
      * @param string $password
+     * @return boolean
      */
     public function findInDatabase($username, $password)
     {
-        die();
+        $query = "Select * from readers where username= \"{$username}\"";
+        $stmt= $this->conn->prepare($query);
+        $stmt->execute();
+        $result = $stmt->fetchAll(PDO::FETCH_BOTH);
+        if (sizeof($result)!= 0 && $result[0][2] == $password) {
```

Comment:
Use named columns selection. For each column you should have a private constant that is used both in the sql query and in the array parsing

Reply:
For the results parsing we can iterate it